Ginger/Ginger/UserControlsLib/TextEditor/ValueExpression/ValueExpressionMockDataEditorPage.xaml.cs

```diff
@@ -0,0 +1,325 @@
+            return Bogus.Database.GetAllLocales().ToList();
+        }
+
+        private static List<string> GetObjectClasses(Assembly assembly, string namespaceName)
+        {
+            Type[] types = assembly.GetTypes();
+            List<string> objclass = types.Where(t => t.Namespace == namespaceName && typeof(DataSet).IsAssignableFrom(t) && t != typeof(DataSet) && t.FullName.Contains(namespaceName))
+                        .Select(x => x.Name)
+                        .ToList();
+            objclass.Add("Randomizer");
+            return objclass;
+        }
+
+        private static List<string> GetMethodsOfType(Type objType)
+        {
+            List<string> methodList = new List<string>();
+
+            if (objType != null)
+            {
+                MethodInfo[] methods = objType.GetMethods(BindingFlags.Public | BindingFlags.Instance | BindingFlags.DeclaredOnly);
+
+                if (methods != null)
+                {
+                    var excludedMethods = GetExcludedMethods(objType.ToString());
+                    var groupedMethods = methods.GroupBy(m => m.Name);
+                    foreach (var methodGroup in groupedMethods)
+                    {
+                        foreach (MethodInfo method in methodGroup)
+                        {
+                            if (excludedMethods.Contains(method.Name))
+                            {
+                                continue;
+                            }
+                            string parameters = string.Join(", ", method.GetParameters().Select(p => FormatParameter(p)));
+                            
+                            string functionString = $"{method.Name}({parameters})";
+                            methodList.Add(functionString);
+                        }
+                    }
+                }
+            }
+
+            return methodList;
+        }
+
+        private static HashSet<string> GetExcludedMethods(string objType)
+        {
+            return objType switch
+            {
+                "Bogus.Randomizer" => new HashSet<string> { "EnumValues", "WeightedRandom", "ArrayElement", "ArrayElements", "ListItem", "ListItems", "ReplaceSymbols" },
+                "Bogus.DataSets.Date" => new HashSet<string> { "BetweenTimeOnly" },
+                _ => new HashSet<string>()
+            };
+        }
+
+        private static string FormatParameter(ParameterInfo parameter)
+        {
+            string paramType = parameter.ParameterType.Name;
+            string paramName = parameter.Name;
+            bool hasDefaultValue = parameter.HasDefaultValue;
+            object defaultValue = hasDefaultValue ? $"{FormatDefaultValue(parameter.DefaultValue)}" : SetDefaultValue(parameter);
+            return $"{defaultValue}";
+            
+        }
+
+        private static object SetDefaultValue(ParameterInfo parameter)
+        {
+            Type parameterType = parameter.ParameterType;
```

Comment:
Please use switch expression here instead of if

Reply:
Done